src/shapes/IText/ITextKeyBehavior.ts

```diff
@@ -295,32 +283,83 @@ export abstract class ITextKeyBehavior<
   }
 
   /**
-   * Copies selected text
+   *
+   * @returns true if text is selected and if the {@link ClipboardEvent#clipboardData} was set
    */
-  copy() {
-    if (this.selectionStart === this.selectionEnd) {
-      //do not cut-copy if no selection
-      return;
-    }
-    const { copyPasteData } = getEnv();
-    copyPasteData.copiedText = this.getSelectedText();
-    if (!config.disableStyleCopyPaste) {
-      copyPasteData.copiedTextStyle = this.getSelectionStyles(
-        this.selectionStart,
-        this.selectionEnd,
-        true
-      );
-    } else {
-      copyPasteData.copiedTextStyle = null;
+  protected setClipboardData(e: ClipboardEvent) {
+    // prevent input event
+    e.preventDefault();
+    const clipboardData = e.clipboardData;
+    if (this.selectionStart === this.selectionEnd || !clipboardData) {
+      return false;
+    }
+    const value = this.getSelectedText();
+    clipboardData.setData('text/plain', value);
+    clipboardData.setData(
+      'application/fabric',
```

Comment:
this should be application/fabric-textstyle in case we decide to copy something else in the future

Reply:
I think all should be uniform `application/fabric` 
I can add a `type: text` to it

Reply:
done